Some edits for wet-labbers to follow

nomadscientist opened the pull request:
I've added some details in various places because running this virtually will be harder than face to face, so it needs to be a bit more hand-holding.

nomadscientist committed:
Some edits for wet-labbers to follow

I've added some details in various places because running this virtually will be harder than face to face, so it needs to be a bit more hand-holding.

pinin4fjords commented:
Hi @nomadscientist , could you remove the .DS_Store please? Don't need that.

pinin4fjords commented on generating_an_expression_matrix/README.md:
```diff
@@ -19,170 +21,355 @@ We'll be using pre-prepared example inputs
 # Introduction
 
-This session will show you the principles of the initial phase of single-cell RNA-seq analysis: generating expression measures in a matrix. We'll concentrate on droplet-based (rather than plate-based) methodology, since this is the process with most differences with respect to conventional approaches developed for bulk RNA-seq.
+This session will show you the principles of the initial phase of
```
Are all the extra line breaks intentional?

nomadscientist commented:
Not at all - my viewer does something weird with line breaks specifically
and those go haywire.

On Tue, May 12, 2020 at 9:36 AM Jonathan Manning <notifications@github.com>
wrote:

> *@pinin4fjords* commented on this pull request.
> ------------------------------
>
> In generating_an_expression_matrix/README.md
> <https://github.com/ebi-gene-expression-group/scrnaseq_training/pull/3#discussion_r423559898>
> :
>
> > @@ -19,170 +21,355 @@ We'll be using pre-prepared example inputs
>
>  # Introduction
>
> -This session will show you the principles of the initial phase of single-cell RNA-seq analysis: generating expression measures in a matrix. We'll concentrate on droplet-based (rather than plate-based) methodology, since this is the process with most differences with respect to conventional approaches developed for bulk RNA-seq.
> +This session will show you the principles of the initial phase of
>
> Are all the extra line breaks intentional?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ebi-gene-expression-group/scrnaseq_training/pull/3#pullrequestreview-409820175>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKUKCSOCXI7U75HP3ERKSCDRREC7RANCNFSM4M6A4YVA>
> .
>

pinin4fjords commented on generating_an_expression_matrix/README.md:
```diff
@@ -19,170 +21,355 @@ We'll be using pre-prepared example inputs
+count as the same gene. In a nutshell, this quantification is used
+because:
+
+ * droplet-based scRNA-seq use short reads, which can make it impossible
```
They don't use especially short reads, they just only sample one end of the transcript.

pinin4fjords commented on generating_an_expression_matrix/README.md:
```diff
@@ -19,170 +21,355 @@ We'll be using pre-prepared example inputs
+
+ * droplet-based scRNA-seq use short reads, which can make it impossible
+to distinguish transcript isoforms; and  
+ * even in full-length single-cell approaches (i.e. smart-seq) counting
```
No, actually quantifying at the transcript level INcreases quantification accuracy at the gene level- see e.g. https://f1000research.com/articles/4-1521, since you're using something like the actual length of the molecules involved.

nomadscientist replied:
Wait, if that's the case why do we use gene-level quantification for smart-seq approaches?

pinin4fjords replied:
People mostly don't find individual transcript readouts very useful- they can be hard to interpret. But again, using transcript-level quantifications to start with improves the gene-level quantification.

pinin4fjords commented on generating_an_expression_matrix/README.md:
```diff
@@ -19,170 +21,355 @@ We'll be using pre-prepared example inputs
+transcripts can lead to inaccurate estimations, largely due to heavy
+length-bias in RNA-seq.
+
+Therefore, scRNA-seq uses gene-level quantification.
```
It's just that because we're only sampling one end of transcripts, we can't really differentially quantify isoforms, unless they start at different positions (I should have been clearer or more coherent on that yesterday!)

nomadscientist replied:
That's fine, this was what I thought initially and I descended deep into a world I don't know enough about re: the length/gene/transcript levels so happy to delete this and just keep the isoforms bit.

pinin4fjords commented:
> Not at all - my viewer does something weird with line breaks specifically and those go haywire.
> […](#)

Could you restore the original line breaks then please? Fixed-width formatting is harder to maintain.

nomadscientist committed:
Deleted DS

Deleting DS, updating the transcript-quantification bits.
Still can't figure out how to make Atom stop automatically Wrapping (despite disabling Wrap Guide) so sort of stuck on line-break issues.

nomadscientist committed:
Update README.md

pinin4fjords commented on generating_an_expression_matrix/README.md:
```diff
@@ -61,98 +61,120 @@ This is what the your history will look like:
 ### 2. Generate a transcript to gene mapping
 
-To generate gene-level quantifications based on transcriptome quantification, Alevin and similar tools require a conversion between transcript and gene identifiers, which we can derive from the annotations available in genome resources such as Ensembl. The transcripts in this list need to match the ones we used in the sequences reference. The study we're working with here added spike-ins to their ...
+Gene-level, rather than transcript-level, quantification is standard in scRNA-seq, which means that that alternatively spliced RNA molecules count as the same gene. In a nutshell, this quantification is used because droplet-based scRNA-seq use short, single-ended reads, which can make it impossible to distinguish transcript isoforms.
```
I know I'm being picky here, but I'm going to pick some more I'm afraid. Single-endendess is not the issue, length is not really the issue. It's the fact that, in contrast to smart-like tech, the reads come from one end of the molecule, rather than being scattered through it. Honestly, I think we should just keep things simple and say that Alevin works from a transcriptome input, but outputs results at the gene level, and therefore needs a mapping.

nomadscientist replied:
Please do be picky, no apologies needed! It should be right. The issue is, even as a wet-labber I get confused at the need for gene vs transcript level quantification and since this is so early in the protocol, people are going to want to know why they are doing this. "The tool works like this" is not the answer here (though I recognise often IS the answer!).  What if I put: 
In a nutshell, this quantification is used because droplet-based scRNA-seq use short reads from one end of the transcript molecule,  which makes it impossible to consistently distinguish transcript isoforms.

pinin4fjords replied:
The thing is, some people do claim that you can do some differential abundance with droplet metrics, so I don't like making categorical statements like that. You probably could have a go in some limited circumstances with short exons, different start points etc, but Alevin doesn't offer that functionality. Pipelines are dictated by the tools available I'm afraid. The fact that differential isoform analysis is difficult with methods like this is sort of a separate issue.

pinin4fjords commented on generating_an_expression_matrix/README.md:
```diff
@@ -61,98 +61,120 @@ This is what the your history will look like:
+> EXERCISE: Which of the 'attributes' in the last column of the GTF files contains the transcript and gene identifiers? Hint: The file is organised such that the last column (headed 'Group') contains a wealth of information in the format: attribute1 "information associated with attribute";attribute2 "information associated with attribute 2" etc.
 
-Now we have combined annotations, we can parse the GTF file using the [rtracklayer](https://bioconductor.org/packages/release/bioc/html/rtracklayer.html) package in R. We've made a convenience tool in Galaxy to wrap that functionality for you- search for 'GTF2GeneList':
+Now we have combined annotations, we can parse the GTF file using the [rtracklayer](https://bioconductor.org/packages/release/bioc/html/rtracklayer.html) package in R. This parsing will give us an index with a list of transcript names and their corresponding gene names for counting. We've made a convenience tool in Galaxy to wrap that functionality for you- search for 'GTF2GeneList':
```
Maybe 'index' is a bit overloaded here? 'Conversion table' instead?

nomadscientist replied:
Yes!

pinin4fjords commented on generating_an_expression_matrix/README.md:
```diff
@@ -61,98 +61,120 @@ This is what the your history will look like:
 Now we have everything we need to run Alevin:
 
- * A Salmon index for our transcriptome
+ * A Salmon index for our transcriptome (transcript name or identifier & sequence)
```
No, index here is the actual binary index structure prepared by Salmon - your adendum is a little confusing I'm afraid.

nomadscientist replied:
OK, so we actually don't have that yet. What we have is the sequence list, which will then be turned into an index by Salmon?

pinin4fjords replied:
Well, this version assumes that a pre-built index is present. If not, we'll have to make a few edits throughout. 

Otherwise the input is the transcriptome sequence itself, not an index.

pinin4fjords commented on generating_an_expression_matrix/README.md:
```diff
@@ -61,98 +61,120 @@ This is what the your history will look like:
+ * protocol
+ * Transcript to gene map file
+Additionally, do select:
+ * Select a reference transcriptome: Built-in index if available (building one from the history index will take a while!)
```
```suggestion
 * Select a reference transcriptome: Built-in index if available (building one from the history transcriptome will take a while!)
```

nomadscientist committed:
Update generating_an_expression_matrix/README.md

Co-authored-by: Jonathan Manning <jmanning@ebi.ac.uk>

pinin4fjords commented on generating_an_expression_matrix/README.md:
```diff
@@ -61,98 +61,120 @@ This is what the your history will look like:
 ![barcode plot from processed barcode counts](barcodes_processed.png)
 
-You should see a completely vertical drop-off where Alevin has trunctated the distribution.
+You should see a completely vertical drop-off where Alevin has trunctated the distribution (excluded any cell barcode that had <10 UMI).
```
```suggestion
You should see a completely vertical drop-off where Alevin has trunctated the distribution.
```
This will be the threshold Alevin picked off the curve, not 10 UMI.

nomadscientist replied:
'Truncated' will be meaningless to a wet-labber. However there is a default threshold of 10 UMI first, and then it will pick a threshold based off the curve, right?

pinin4fjords replied:
Happy for you to use a synonym for truncated- but the reason the plot looks as it does is because Alevin has picked a threshold, not due to any parameter we input.

nomadscientist replied:
OK! But to be clear, the 'default threshold: 10' is not doing anything in this scenario?

pinin4fjords replied:
That's just the bare minimum, in most cases Alevin will be picking a point further up the distribution than that.

nomadscientist committed:
Update README.md

nomadscientist committed:
Merge branch 'master' of https://github.com/nomadscientist/scrnaseq_training

nomadscientist committed:
Update README.md

nomadscientist committed:
Update README.md

Tricky rather than impossible, avoiding definites!

nomadscientist commented:
I think I've addressed these issues in the later commits (hopefully).

nomadscientist committed:
Alevin not searchable

So included a note on searching for it

pinin4fjords commented:
Right, I'm going to merge these (thanks for the contributions), and then I might make a couple of small tweaks myself.
